lib/src/model/enum.dart

```diff
@@ -8,31 +8,8 @@ import 'package:dartdoc/src/model/model.dart';
 import 'package:dartdoc/src/model_utils.dart' as model_utils;
 import 'package:dartdoc/src/render/enum_field_renderer.dart';
 
-/// The [Enum] class only inherits from [InheritingContainer] because declared
-/// `enum`s inherit methods from [Object].  It can't actually participate
-/// meaningfully in other inheritance or have class modifiers.
-class Enum extends InheritingContainer
-    with Constructable, TypeImplementing, MixedInTypes {
-  @override
-  final EnumElement element;
-
-  Enum(this.element, super.library, super.packageGraph);
-
-  @override
-  late final List<ModelElement> allModelElements = [
-    ...super.allModelElements,
-    ...constructors,
-  ];
-
-  @override
-  late final List<InheritingContainer> inheritanceChain = [
-    this,
-    for (var container in mixedInElements.reversed)
-      ...container.inheritanceChain,
-    for (var container in superChain.modelElements)
-      ...container.inheritanceChain,
-    ...interfaceElements.expandInheritanceChain,
-  ];
+class Enum extends Class {
```

Comment:
I know there's shared behaviour that we want, but this feels a little weird semantically. Could we rename Class to Interface or something similar? I feel like we're making the hierarchy even more confusing.

Reply:
I could back out this change, if it makes the hierarchy more confusing. I wonder if a cleaner hierarchy would be:

* Interface, which maybe aligns with analyzer's InterfaceElement
* Enum extends Interface, aligning with analyzer's EnumElement
* Class extends Interface, aligning with analyzer's ClassElement
* Mixin extends Interface, aligning with analyzer's MixinElement

I'll look into this.

Reply:
OK so that already existed, it was InheritingContainer 😛  So I just backed out this part of the change, and we'll leave the inheritance like it is. It only reduced some duplication, but at the expense of a correct hierarchy is bad.

Reply:
Sounds good to me, thanks.
I was just afraid it would become another situation where we're like "wait what does this mean?"